botorch/acquisition/multi_objective/monte_carlo.py

```diff
@@ -752,3 +759,110 @@ def forward(self, X: Tensor) -> Tensor:
+                Defaults to `IdentityMultiOutputObjective()`.
+            constraints: A list of callables, each mapping a Tensor of dimension
+                `sample_shape x batch-shape x q x m` to a Tensor of dimension
+                `sample_shape x batch-shape x q`, where negative values imply
+                feasibility. The acqusition function will compute expected feasible
+                hypervolume.
+            X_pending: A `batch_shape x m x d`-dim Tensor of `m` design points that have
+                points that have been submitted for function evaluation but have not yet
+                been evaluated. Concatenated into `X` upon forward call. Copied and set
+                to have no gradient.
+            cost_call: A callable cost function mapping a Tensor of dimension
+                `batch_shape x q x d` to a cost Tensor of dimension
+                `batch_shape x q x m`.
+                Defaults to an AffineCostModel with C(s)=1+s.
+            eta: The temperature parameter for the sigmoid function used for the
+                differentiable approximation of the constraints.
+        """
+
+        if len(ref_point) != partitioning.num_outcomes:
+            raise ValueError(
+                "The length of the reference point must match the number of outcomes. "
+                f"Got ref_point with {len(ref_point)} elements, but expected "
+                f"{partitioning.num_outcomes}."
+            )
+        ref_point = torch.as_tensor(
+            ref_point,
+            dtype=partitioning.pareto_Y.dtype,
+            device=partitioning.pareto_Y.device,
+        )
+
+        super(qExpectedHypervolumeImprovement, self).__init__(
+            model=model,
+            sampler=sampler,
+            objective=objective,
+            constraints=constraints,
+            X_pending=X_pending,
+        )
+        self.eta = eta
+        self.register_buffer("ref_point", ref_point)
+        cell_bounds = partitioning.get_hypercell_bounds()
+        self.register_buffer("cell_lower_bounds", cell_bounds[0])
+        self.register_buffer("cell_upper_bounds", cell_bounds[1])
+        self.q_out = -1
+        self.q_subset_indices = BufferDict()
+        self.cost_call = cost_call
+
+        if self.cost_call is None:
+            cost_model = AffineFidelityCostModel(
+                fidelity_weights={-1: 1.0}, fixed_cost=1.0
+            )
+        else:
+            cost_model = GenericDeterministicModel(cost_call)
+        cost_aware_utility = InverseCostWeightedUtility(cost_model=cost_model)
+        self.cost_aware_utility = cost_aware_utility
+
+    @concatenate_pending_points
+    @t_batch_mode_transform()
+    def forward(self, X: Tensor) -> Tensor:
+        posterior = self.model.posterior(X)
+        samples = self.sampler(posterior)
+        hv_gain = self._compute_qehvi(samples=samples, X=X)
+        ig = self.cost_aware_utility(X=X, deltas=hv_gain)
```

Comment:
What is ig here? Information gain? It seems like cost-weighted qehvi is more appropriate